propagation/b3/http.go

```diff
@@ -31,8 +55,15 @@ func ExtractHTTP(r *http.Request) propagation.Extractor {
 			parentSpanIDHeader = r.Header.Get(ParentSpanID)
 			sampledHeader      = r.Header.Get(Sampled)
 			flagsHeader        = r.Header.Get(Flags)
+			singleHeader       = r.Header.Get(Context)
 		)
 
+		if singleHeader != "" {
+			if sc, err := ParseSingleHeader(singleHeader); err == nil {
```

Comment:
If we error parsing single header then we forward this concern to the multi header.

Reply:
almost... if ParseHeaders fails too, I'd rather return the error from the single header parser (if single header was available)

Reply:
Makes sense.

Reply:
I also added more tests for checking this fallback too.